pkg/testrunner/result/collect.go

```diff
@@ -63,6 +65,27 @@ func (c *Collector) Collect(log logr.Logger, tmClient kubernetes.Interface, name
 			}
 		}
 
+		// upload testrun status to github component
+		if cfg.GithubComponentForStatus != "" {
```

Comment:
maybe we can move this check into a separate function as proposed by you?
now I see that it is more code than I thought so it would make sense to factor it out.
Then we can also get rid of these else nesting